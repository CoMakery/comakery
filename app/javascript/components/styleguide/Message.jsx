import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Icon from './Icon'

class Message extends React.Component {
  constructor(props) {
    super(props)
    this.hide = this.hide.bind(this)
    this.state = {
      hidden: false
    }
  }

  hide() {
    this.setState({
      hidden: true
    })
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ hidden: false })
  }

  render() {
    const {className, severity, text, ...other} = this.props

    const classnames = classNames(
      'message',
      className
    )

    return (
      <React.Fragment>
        {!this.state.hidden &&
          <div className="message--wrapper--block">
            <div className="message--wrapper--inline">
              <div className={classnames} {...other}>
                <span className="message--icon">
                  { severity === 'warning' &&
                    <Icon name="WARNING.svg" width={21} />
                  }
                  { severity === 'error' &&
                    <Icon name="ALERT.svg" width={21} />
                  }
                </span>
                <span className={`message--text__${severity}`}>
                  {text}
                </span>
                <span className="message--close-icon" onClick={this.hide} >
                  <Icon name="iconCloseCopy.svg" width={24} />
                </span>
              </div>
            </div>
          </div>
        }
      </React.Fragment>
    )
  }
}

Message.propTypes = {
  className: PropTypes.string,
  severity : PropTypes.string,
  text     : PropTypes.string
}
Message.defaultProps = {
  className: '',
  severity : 'warning',
  text     : 'warning text'
}
export default Message
