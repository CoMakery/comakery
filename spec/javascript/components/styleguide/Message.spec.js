import React from 'react'
import { shallow } from 'enzyme'
import Message from 'components/styleguide/Message'

describe('Message', () => {
  it('renders correctly without props', () => {
    const wrapper = shallow(<Message />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.exists('.message')).toBe(true)
    expect(wrapper.exists('.message--icon')).toBe(true)
    expect(wrapper.exists('.message--text__warning')).toBe(true)
    expect(wrapper.exists('.message--close-icon')).toBe(true)
    expect(wrapper.find('.message--text__warning').text()).toBe('warning text')
  })

  it('renders correctly with custom className', () => {
    const wrapper = shallow(<Message className="__test" />)

    expect(wrapper.exists('.message.__test')).toBe(true)
  })

  it('renders correctly with custom severity', () => {
    const wrapper = shallow(<Message severity="error" />)

    expect(wrapper.exists('.message--text__error')).toBe(true)
  })

  it('renders correctly with custom text', () => {
    const wrapper = shallow(<Message text="test" />)

    expect(wrapper.find('.message--text__warning').text()).toBe('test')
  })

  it('gets hidden on close click', () => {
    const wrapper = shallow(<Message />)

    wrapper.find('.message').find('.message--close-icon').simulate('click')
    expect(wrapper.exists('.message')).toBe(false)
  })
})
