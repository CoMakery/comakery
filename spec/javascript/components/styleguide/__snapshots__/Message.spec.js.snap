// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders correctly without props 1`] = `
<Fragment>
  <div
    className="message--wrapper--block"
  >
    <div
      className="message--wrapper--inline"
    >
      <div
        className="message"
      >
        <span
          className="message--icon"
        >
          <Icon
            className=""
            name="Atoms---Icons-System-Heart.png"
            width={21}
          />
        </span>
        <span
          className="message--text__warning"
        >
          warning text
        </span>
        <span
          className="message--close-icon"
          onClick={[Function]}
        >
          <Icon
            className=""
            name="Atoms---Icons-System-Heart.png"
            width={21}
          />
        </span>
      </div>
    </div>
  </div>
</Fragment>
`;
